modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -328,14 +334,7 @@ public static ClassLoader igniteClassLoader() {
         if (ldr == null)
             ldr = igniteClassLoader;
 
-        ConcurrentMap<String, Class<?>> ldrMap = classCache.get(ldr);
-
-        if (ldrMap == null) {
-            ConcurrentMap<String, Class<?>> old = classCache.putIfAbsent(ldr, ldrMap = new ConcurrentHashMap<>());
-
-            if (old != null)
-                ldrMap = old;
-        }
+        ConcurrentMap<String, Class<?>> ldrMap = classCache.computeIfAbsent(ldr, k -> new ConcurrentHashMap<>());
```

Comment:
> Is there a reason for this change other than code simplicity?
no
> Sadly, computeIfAbsent is a blocking method
I think we had a discussion about this a while ago and we found out that this is not entirely true, at least for Java 9+

Reply:
I also think we had it but results weren't that clear. It seems like Java9+ CHashMap works exactly the same. Quick googling revealed that computeIfAbsent can be a bottleneck, I would recommend reverting this particular fix.

Reply:
https://bugs.openjdk.java.net/browse/JDK-8161372 - is this related? Which article showed you that it might be a bottleneck? Anyway, I will rollback this change since, strictly speaking, it is not required by this PR

Reply:
Sort of, you're right, it's better with 9+